### PR TITLE
COCICCS-5681 API: Create AWS Roles

### PR DESCRIPTION
**Pull request by msivatmob**

### CODE REVIEW

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SelfSupportService.java [hunk 1]
```diff
@@ -1252,4 +1252,25 @@ else if (response!=null && HttpStatus.NOT_FOUND.equals(response.getHttpstatus())
 		return ResponseEntity.status(response==null?null:response.getHttpstatus()).body(response.getResponse());	
 	}
 	
+	public ResponseEntity<String> createAwsec2Role(UserDetails userDetails, String token, AWSLoginRole awsLoginRole) throws TVaultValidationException {
```
Comment (alexrajan3) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SelfSupportService.java:1255]: Add method comments

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SelfSupportService.java [hunk 2]
```diff
@@ -1252,4 +1252,25 @@ else if (response!=null && HttpStatus.NOT_FOUND.equals(response.getHttpstatus())
...
+		if (userDetails.isAdmin()) {
+			 accesstoken = userDetails.getClientToken();
+		}
+		else {
+			accesstoken = userDetails.getSelfSupportToken();
+		}
+		return awsAuthService.createRole(accesstoken, awsLoginRole, userDetails);
+	}
+
+	public ResponseEntity<String> createAwsiamRole(UserDetails userDetails, String token, AWSIAMRole awsiamRole) throws TVaultValidationException {
```
Comment (alexrajan3) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SelfSupportService.java:1266]: Add method comments

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/v2/controller/SelfSupportController.java [hunk 3]
```diff
@@ -255,9 +255,8 @@
 		UserDetails userDetails = (UserDetails) ((HttpServletRequest) request).getAttribute("UserDetails");
 		return selfSupportService.deleteApproleFromSDB(userDetails, token, safeAppRoleAccess);
 	}
-
 	/**
-	 * Method to create an aws app role
+	 * Method to create an ec2 aws  role
```
Comment (alexrajan3) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/v2/controller/SelfSupportController.java:259]: Change the text "ec2 aws" to "AWS EC2"

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/v2/controller/SelfSupportController.java [hunk 4]
```diff
@@ -496,4 +494,28 @@
 		UserDetails userDetails = (UserDetails) (request).getAttribute("UserDetails");
 		return selfSupportService.listRoles(token,userDetails);
 	}
+	/**
+	 * Method to create an ec2 aws  role in new UI
```
Comment (alexrajan3) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/v2/controller/SelfSupportController.java:498]: Change the text "ec2 aws" to "AWS EC2"

File: tvaultapi/src/main/resources/swaggernotes.properties [hunk 5]
```diff
@@ -902,4 +902,11 @@ AzureServicePrincipalAccountsController.removeApproleFromAzureSvcAcc.notes=<b>Re
 <b>access: </b>Access cannot be null or empty. Access can have values read, rotate and deny.
 
 SecretControllerV2.getSecretCount.value=To get total secret count in T-Vault
-SecretControllerV2.getSecretCount.notes=To get total secret count in T-Vault
\ No newline at end of file
+SecretControllerV2.getSecretCount.notes=To get total secret count in T-Vault
+
+SelfSupportController.createAwsec2Role.value=To create AwsEc2 roles in New UI
```
Comment (alexrajan3) [inline tvaultapi/src/main/resources/swaggernotes.properties:907]: Change the text "AwsEc2" to "AWS EC2"

File: tvaultapi/src/main/resources/swaggernotes.properties [hunk 6]
```diff
@@ -902,4 +902,11 @@ AzureServicePrincipalAccountsController.removeApproleFromAzureSvcAcc.notes=<b>Re
...
+SelfSupportController.createAwsec2Role.notes=To create AwsEc2 roles in New UI
```
Comment (alexrajan3) [inline tvaultapi/src/main/resources/swaggernotes.properties:908]: Change the text "AwsEc2" to "AWS EC2"

File: tvaultapi/src/main/resources/swaggernotes.properties [hunk 7]
```diff
@@ -902,4 +902,11 @@ AzureServicePrincipalAccountsController.removeApproleFromAzureSvcAcc.notes=<b>Re
...
+SelfSupportController.createAwsiamRole.value=To create AwsIam roles in new UI
```
Comment (alexrajan3) [inline tvaultapi/src/main/resources/swaggernotes.properties:909]: Change the text "AwsIam" to "AWS IAM"

File: tvaultapi/src/main/resources/swaggernotes.properties [hunk 8]
```diff
@@ -902,4 +902,11 @@ AzureServicePrincipalAccountsController.removeApproleFromAzureSvcAcc.notes=<b>Re
...
+SelfSupportController.createAwsiamRole.notes=To create AwsIam roles in new UI
```
Comment (alexrajan3) [inline tvaultapi/src/main/resources/swaggernotes.properties:910]: Change the text "AwsIam" to "AWS IAM"

**Review by alexrajan3 — CHANGES_REQUESTED** [review]
Please correct the below changes

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SelfSupportService.java (hunk 1 above)
Comment (msivatmob) [reply tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SelfSupportService.java:1263]: Method comments added.

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SelfSupportService.java (hunk 2 above)
Comment (msivatmob) [reply tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SelfSupportService.java:1281]: Method comments added.

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/v2/controller/SelfSupportController.java (hunk 3 above)
Comment (msivatmob) [reply tvaultapi/src/main/java/com/tmobile/cso/vault/api/v2/controller/SelfSupportController.java]: Text changed to AWS EC2.

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/v2/controller/SelfSupportController.java (hunk 4 above)
Comment (msivatmob) [reply tvaultapi/src/main/java/com/tmobile/cso/vault/api/v2/controller/SelfSupportController.java]: Text changed to AWS EC2.

File: tvaultapi/src/main/resources/swaggernotes.properties (hunk 6 above)
Comment (msivatmob) [reply tvaultapi/src/main/resources/swaggernotes.properties]: Text changed to AWS EC2.

File: tvaultapi/src/main/resources/swaggernotes.properties (hunk 7 above)
Comment (msivatmob) [reply tvaultapi/src/main/resources/swaggernotes.properties]: Text changed to AWS IAM.

File: tvaultapi/src/main/resources/swaggernotes.properties (hunk 8 above)
Comment (msivatmob) [reply tvaultapi/src/main/resources/swaggernotes.properties]: Text changed to AWS IAM.

File: tvaultapi/src/main/resources/swaggernotes.properties (hunk 5 above)
Comment (msivatmob) [reply tvaultapi/src/main/resources/swaggernotes.properties]: Text changed to AWS EC2.

**Review by alexrajan3 — APPROVED** [review]

**Pull request merged**